frontend/__tests__/utils/zod-utils.test.ts

```diff
@@ -0,0 +1,59 @@
+import { describe, expect, it } from 'vitest';
+
+import { transformFlattenedError } from '~/utils/zod.utils';
```

Comment:
Since the Zod utils is only used by `action(..)`/server-side and to follow existing naming util file naming conventions:

```suggestion
import { transformFlattenedError } from '~/utils/zod-utils.server';
```

Reply:
Fair point. Fixed